cli/pa_gather_command.go

```diff
@@ -0,0 +1,709 @@
+    ······Save the stream info
+*/
+func (c *paGatherCmd) gather(_ *fisk.ParseContext) error {
+
+	nc, err := newNatsConn("", natsOpts()...)
+	if err != nil {
+		return err
+	}
+	defer nc.Close()
+
+	// If no output path is specified, create one
+	if c.archiveFilePath == "" {
+		c.archiveFilePath = filepath.Join(os.TempDir(), "archive.zip")
+	}
+
+	// Gathering prints messages to stdout, but they are also written into this buffer.
+	// A copy of the output is included in the archive itself.
+	var captureLogBuffer bytes.Buffer
+	c.captureLogWriter = &captureLogBuffer
+
+	// Create an archive writer
+	aw, err := archive.NewWriter(c.archiveFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create archive: %w", err)
+	}
+	defer func() {
+		// Add the output of this command (so far) to the archive as additional log artifact
+		if c.captureLogWriter != nil {
+			err = aw.AddCaptureLog(bytes.NewReader(captureLogBuffer.Bytes()))
+			if err != nil {
+				fmt.Printf("Failed to add capture log artifact: %s", err)
+			}
+			c.captureLogWriter = nil
+		}
+
+		err := aw.Close()
+		if err != nil {
+			fmt.Printf("Failed to close archive: %s", err)
+		}
+		fmt.Printf("📁 Archive created at: %s\n", c.archiveFilePath)
+	}()
+
+	// Collect some runtime metadata and store it as additional artifact
+	{
+		username := "?"
+		currentUser, err := user.Current()
+		if err != nil {
+			c.logWarning("Failed to capture username: %s", err)
+		} else {
+			username = fmt.Sprintf("%s (%s)", currentUser.Username, currentUser.Name)
+		}
+
+		err = aw.AddCaptureMetadata(&struct {
+			Timestamp              time.Time `json:"capture_timestamp"`
+			ConnectedServerName    string    `json:"connected_server_name"`
+			ConnectedServerVersion string    `json:"connected_server_version"`
+			ConnectURL             string    `json:"connect_url"`
+			UserName               string    `json:"user_name"`
+			CLIVersion             string    `json:"cli_version"`
+		}{
+			Timestamp:              time.Now(),
+			ConnectedServerName:    nc.ConnectedServerName(),
+			ConnectedServerVersion: nc.ConnectedServerVersion(),
+			ConnectURL:             nc.ConnectedUrl(),
+			UserName:               username,
+			CLIVersion:             Version,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to save capture metadata: %w", err)
+		}
+	}
+
+	// CustomServerAPIResponse is a modified version of server.ServerAPIResponse that inhibits deserialization of the
+	// `data` field by using `json.RawMessage`. This is necessary because deserializing into a generic object (i.e. map)
+	// can cause loss of precision for large numbers.
+	type CustomServerAPIResponse struct {
+		Server *server.ServerInfo `json:"server"`
+		Data   json.RawMessage    `json:"data,omitempty"`
+		Error  *server.ApiError   `json:"error,omitempty"`
+	}
+
+	/*
+		1. Servers discovery
+	*/
+
+	// Discover servers by broadcasting a PING and then collecting responses
+	var serverInfoMap = make(map[string]*server.ServerInfo)
```

Comment:
re: `currentActiveServers()`

Given the nature of the tool, I think it's best to ping and wait for all answers rather than trust the first server that responds.

I'm thinking: one misconfigured server responds fast saying "there are 0 other servers", and then we miss out on capturing everyone else.